python/kvikio/_lib/libnvcomp_ll.pyx

```diff
@@ -376,6 +526,92 @@ class nvCompBatchAlgorithmLZ4(nvCompBatchAlgorithm):
 
         return (err, temp_bytes)
 
+    def get_decompress_size(
+        self,
+        comp_chunks,
+        comp_chunk_sizes,
+        stream,
+    ):
+        if not self.has_header:
+            return super().get_decompress_size(
+                comp_chunks,
+                comp_chunk_sizes,
+                stream,
+            )
+
+        assert comp_chunks.shape == comp_chunk_sizes.shape
+        batch_size = len(comp_chunks)
+
+        # uncomp_chunk_sizes is uint32 array to match the type in LZ4 header.
+        uncomp_chunk_sizes = cupy.empty(batch_size, dtype=cupy.uint32)
+
+        psize = to_ptr(uncomp_chunk_sizes)
+        for i in range(batch_size):
+            # Get pointer to the header and copy the data.
+            memcpyAsync(
+                psize,
+                <uintptr_t>comp_chunks[i],
+                sizeof(uint32_t),
+                cudaMemcpyKind.cudaMemcpyDeviceToDevice,
+                stream.ptr
+            )
+            psize += sizeof(uint32_t)
```

Comment:
Perhaps more esoteric, but I wonder if is worthwhile using `cudaMemcpy2DAsync` instead here.

Which would be, I think

```
rows, columns = comp_chunks.shape
cudaMemcpy2DAsync(psize, sizeof(uint32_t), <uintptr_t>comp_chunks[0], 
                  columns, sizeof(uint32_t), rows, 
                  cudaMemcpyKind.cudaMemcpyDeviceToDevice, stream.ptr)
```
Then there's no need for the loop over batches.

WDYT?

Reply:
This is a good idea, but in this particular case it won't work: the chunks that are passed to `get_decompress_size` will be of different sizes (unless the source data was identical set of bytes or something like that), so it's not really a 2D array.

Reply:
Ah ok, scratch that then.